src/conf.c

```diff
@@ -386,9 +386,15 @@ int ConfGetInt(const char *name, intmax_t *val)
     if (ConfGet(name, &strval) == 0)
         return 0;
 
+    if (strval == NULL) {
+        SCLogError(SC_ERR_INVALID_YAML_CONF_ENTRY, "malformed integer value "
+                "for %s: NULL", name);
+        return 0;
+    }
+
     errno = 0;
     tmpint = strtoimax(strval, &endptr, 0);
-    if (strval[0] == '\0' || *endptr != '\0') {
+    if (*endptr != '\0') {
```

Comment:
why is the first condition removed here?

Reply:
Because this condition is checked before strtoimax() now so that strval can't be NULL when it's used by strtoimax. Since strtoimax does not change strval, another check of strval would be useless.

Reply:
The checks are not the same though. One checks that the ptr != NULL, the other checks that the first char is not \0

Reply:
Shouldn't the strval[0] check be moved up with strval == NULL so you don't even bother calling strtoimax?

Reply:
How embarrassing. Of course you are right with the first char. I put the strval[0]-check back to it's place, but if you want I would also move it up to the NULL-check before strtoimax.